core/storage/base_model/gae_models.py

```diff
@@ -922,8 +944,7 @@ def delete_multi(
                     0,
                     len(all_models_keys),
                     feconf.MAX_NUMBER_OF_OPS_IN_TRANSACTION):
-                transaction_services.run_in_transaction(
-                    datastore_services.delete_multi,
```

Comment:
Here is the mistake, you replace method in `datastore_services` services with method in `BaseModel`, if you look at `delete_multi` in `datastore_services` it is bit different from `delete_multi` in `BaseModel`.

Reply:
Now I have a clear understanding of what went wrong, thanks. Just making sure so I don't mess this up, for fixing this should I simply go for ```datastore_services.delete_multi(
                    all_models_keys[
                        i:i + feconf.MAX_NUMBER_OF_OPS_IN_TRANSACTION])``` or follow a similar approach like ```put_multi``` and declare a ```delete_multi_transactional``` in ```gae_datastore_services``` as well and use that here.

Reply:
Thanks, working on it.

Reply:
Just one last question, should I bother making changes to ```gae_datastore_services.py``` and start by defining a transactional version for ```delete_multi``` there, or just limit my work to ```gae_models.py```